sdk/kfserving/api/kf_serving_api.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2019 The Kubeflow Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from kubernetes import client, config, watch
+from kfserving.constants import constants
+
+class KFServingApi(object):
+    """KFServing Apis."""
+
+    def __init__(self):
+        config.load_kube_config()
+        self.api_instance = client.CustomObjectsApi()
+    
+    def deploy(self, kfservice, namespace='default'):
```

Comment:
I think namespace field is already on kfservice metadata, we can have some defaulting logic if namespace is not set then get the namespace from current kube context?

Reply:
Great comments! So updated logics are as following **if the namespace is not defined** for APIs:
1. `default_namespace`: If in k8s cluster, get the current namespace, if not, set the namespace to 'default'.
2. For `deploy` and `patch`, try to get the namespace from defined kfservice firstly, if did not get that, then will get `default_namespace` in step #1. 
3. For `get` and `delete`, since no kfservice defined, so just derectly use `default_namespace` in step #1.